docs/spec_draft.md

```diff
@@ -1632,6 +1633,60 @@ where `i[d] = j[permutation[d]]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.while
+
+### Semantics
+Produces the output from executing `body` function zero or more times till the
+`cond` function outputs `true`. More formally, `body` function is executed once
+every time the `cond` function returns `true` and all execution stops once
+`cond` function returns `false`.
```

Comment:
We need to say something about how the data flow across the two region blocks. For example, the two regions arguments are initialed with the inputs of `stablehlo.while` in the first iteration. In the subsequent iterations the arguments are fed by the return value of the `body` region.

We can use https://mlir.llvm.org/docs/Dialects/SCFDialect/#scfwhile-mlirscfwhileop as  a reference.

Reply:
Yes, let me update the semantics

Reply:
Does this look better now?